plugin/trino-hive/src/test/java/io/trino/plugin/hive/coercions/TestBooleanCoercer.java

```diff
@@ -53,11 +53,80 @@ public void testBooleanToLowerBoundedVarchar()
                 .hasMessageContaining("Varchar representation of false exceeds varchar(1) bounds");
     }
 
+    @Test
+    public void testVarcharToBoolean()
+    {
+        // Valid false values
+        assertVarcharToBooleanCoercion("FALSE", false);
+        assertVarcharToBooleanCoercion("OFF", false);
+        assertVarcharToBooleanCoercion("NO", false);
+        assertVarcharToBooleanCoercion("0", false);
+        assertVarcharToBooleanCoercion("", false);
```

Comment:
What is the behaviour for space `" "` in Hive? Can we add test for this?

Reply:
I have replaced 'ON` with ' ' so as to verify the behavior